File: frontend/src/api/strategy.ts
/**
 * 策略相关 API
 */
import { request } from './request'

export interface Strategy {
  id: number
  user_id: number
  name: string
  description?: string
  code: string
  params?: Record<string, any>
  status: string
  is_backtest: boolean
  created_at: string
  updated_at?: string
}

export interface StrategyCreate {
  name: string
  description?: string
  code: string
  params?: Record<string, any>
}

export interface StrategyUpdate {
  name?: string
  description?: string
  code?: string
  params?: Record<string, any>
  status?: string
}

export interface BacktestCreate {
  strategy_id: number
  start_date: string
  end_date: string
  initial_capital?: number
}

export interface BacktestResult {
  id: number
  strategy_id: number
  start_date: string
  end_date: string
  initial_capital: number
  final_capital?: number
  total_return?: number
  sharpe_ratio?: number
  max_drawdown?: number
  win_rate?: number
  total_trades?: number
  result_data?: Record<string, any>
  created_at: string
}

// 获取策略列表
export const getStrategies = (params?: { skip?: number; limit?: number }) => {
  return request.get<Strategy[]>('/strategies/', { params })
}

// 获取策略详情
export const getStrategy = (id: number) => {
  return request.get<Strategy>(`/strategies/${id}`)
}

// 创建策略
export const createStrategy = (data: StrategyCreate) => {
  return request.post<Strategy>('/strategies/', data)
}

// 更新策略
export const updateStrategy = (id: number, data: StrategyUpdate) => {
  return request.put<Strategy>(`/strategies/${id}`, data)
}

// 删除策略
export const deleteStrategy = (id: number) => {
  return request.delete(`/strategies/${id}`)
}

// 启动策略
export const startStrategy = (id: number) => {
  return request.post(`/strategies/${id}/start`)
}

// 停止策略
export const stopStrategy = (id: number) => {
  return request.post(`/strategies/${id}/stop`)
}

// 创建回测
export const createBacktest = (data: BacktestCreate) => {
  return request.post<BacktestResult>('/strategies/backtest', data)
}

// 获取回测结果
export const getBacktest = (id: number) => {
  return request.get<BacktestResult>(`/strategies/backtest/${id}`)
}

File: backend/app/schemas/strategy.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional, Dict, Any


class StrategyBase(BaseModel):
    """策略基础模型"""
    name: str = Field(..., min_length=1, max_length=100)
    description: Optional[str] = None
    code: str
    params: Optional[Dict[str, Any]] = None


class StrategyCreate(StrategyBase):
    """策略创建模型"""
    pass


class StrategyUpdate(BaseModel):
    """策略更新模型"""
    name: Optional[str] = Field(None, min_length=1, max_length=100)
    description: Optional[str] = None
    code: Optional[str] = None
    params: Optional[Dict[str, Any]] = None
    status: Optional[str] = None


class StrategyResponse(StrategyBase):
    """策略响应模型"""
    id: int
    user_id: int
    status: str
    is_backtest: bool
    created_at: datetime
    updated_at: Optional[datetime] = None
    
    class Config:
        from_attributes = True


class BacktestCreate(BaseModel):
    """回测创建模型"""
    strategy_id: int
    start_date: datetime
    end_date: datetime
    initial_capital: float = 100000.0


class BacktestResponse(BaseModel):
    """回测响应模型"""
    id: int
    strategy_id: int
    start_date: datetime
    end_date: datetime
    initial_capital: float
    final_capital: Optional[float] = None
    total_return: Optional[float] = None
    sharpe_ratio: Optional[float] = None
    max_drawdown: Optional[float] = None
    win_rate: Optional[float] = None
    total_trades: Optional[int] = None
    result_data: Optional[Dict[str, Any]] = None
    created_at: datetime
    
    class Config:
        from_attributes = True
